packages/api/cms-api/src/common/errors/exception.filter.ts

```diff
@@ -0,0 +1,62 @@
+import {
+    ArgumentsHost,
+    BadRequestException,
+    Catch,
+    ExceptionFilter as NestExceptionFilter,
+    HttpException,
+    InternalServerErrorException,
+} from "@nestjs/common";
+import { ErrorHttpStatusCode } from "@nestjs/common/utils/http-error-by-code.util";
+import { Response } from "express";
+
+import { CometException } from "./comet.exception";
+import { CometValidationException } from "./validation.exception";
+
+@Catch()
+export class ExceptionFilter implements NestExceptionFilter {
+    constructor(private readonly debug: boolean) {}
+
+    catch(exception: Error, host: ArgumentsHost) {
+        let statusCode: ErrorHttpStatusCode;
+        let returnedError: Error;
+
+        if (exception instanceof CometException) {
+            const errorObject: Record<string, unknown> = {
+                statusCode: 400,
+                message: exception.message,
+                error: exception.constructor.name,
+                validationErrors: [],
+            };
+
+            if (exception instanceof CometValidationException) {
+                errorObject.validationErrors = exception.errors;
+            }
+
+            statusCode = 400;
+            returnedError = new BadRequestException(errorObject);
+        } else if (exception instanceof HttpException) {
+            statusCode = exception.getStatus();
+            returnedError = exception;
+        } else {
+            returnedError = this.debug ? exception : new InternalServerErrorException();
+            statusCode = "getStatus" in returnedError && typeof returnedError.getStatus === "function" ? returnedError.getStatus() : 500;
+            console.error(exception); // Log for debugging
```

Comment:
we might use a [Logger](https://docs.nestjs.com/techniques/logger) here to have a timestamp

Reply:
d78df1196e9960c808751fbccbe1f0831336a773